app/src/main/kotlin/com/google/samples/apps/nowinandroid/ui/interests2pane/InterestsListDetailScreen.kt

```diff
@@ -76,17 +83,38 @@ internal fun InterestsListDetailScreen(
     selectedTopicId: String?,
     onTopicClick: (String) -> Unit,
 ) {
-    val listDetailNavigator = rememberListDetailPaneScaffoldNavigator()
+    val listDetailNavigator = rememberListDetailPaneScaffoldNavigator(
+        initialDestinationHistory = listOfNotNull(
+            ThreePaneScaffoldDestinationItem(ListDetailPaneScaffoldRole.List),
+            ThreePaneScaffoldDestinationItem<Nothing>(ListDetailPaneScaffoldRole.Detail).takeIf {
+                selectedTopicId != null
+            },
+        ),
+    )
     BackHandler(listDetailNavigator.canNavigateBack()) {
         listDetailNavigator.navigateBack()
     }
 
-    val nestedNavController = rememberNavController()
+    var nestedNavHostStartDestination by remember {
+        mutableStateOf(selectedTopicId?.let(::createTopicRoute) ?: TOPIC_ROUTE)
+    }
+    var nestedNavKey by remember { mutableStateOf(UUID.randomUUID()) }
+    val nestedNavController = key(nestedNavKey) {
```

Comment:
I think you can use `nestedNavHostStartDestination` instead and remove `nestedNavKey`? (They are both only changed at the same time.)

Reply:
There's an edge case where that doesn't work if `nestedNavHostStartDestination` doesn't change:

On a compact screen, where just one pane is showing, if we tap on "Compose", the `nestedNavHostStartDestination` will be the "Compose" starting route. If we then tap on the "Architecture" chip from an article on the "Compose" details pane, we'll navigate with the `nestedNavController` to "Architecture". If we then navigate back, we'll end up back on the list screen. If we then click on "Compose" again, the `nestedNavHostStartDestination` won't change, so we won't recreate the `nestedNavController` and nested `NavHost`, so we'll end up at the old state of "Architecture".